wrapanapi/systems/ec2.py

```diff
@@ -832,6 +832,36 @@ def list_templates(self, executable_by_me=True, owned_by_me=True, public=False):
         return [EC2Image(system=self, raw=self.ec2_resource.Image(image["ImageId"]))
                 for image in images]
 
+    def filter_free_images(self, image_list=None):
+        """
+        Returns images which don't have a VM associated to it
+
+        Args:
+            image_list (list): List of images
+        """
+        occupied_images = []
+        vm_list = self.list_vms()
+
+        if len(vm_list) == 0:
+            # No VMs using the images, images are free
+            return image_list
+
+        for vm in vm_list:
+            if vm.raw.image_id in image_list:
+                occupied_images.append(vm.raw.image_id)
+        free_images = list(set(image_list) - set(occupied_images))
+        return free_images
```

Comment:
```suggestion
        free_images = []
        vm_list = self.list_vms()

        if len(vm_list) == 0:
            # No VMs using the images, images are free
            return image_list

        for vm in vm_list:
            if vm.raw.image_id not in image_list:
                free_images.append(vm.raw.image_id)
        return free_images
```

Go opposite, find free images only :)